guides/hack/08-arrays-and-collections/01-introduction.md

```diff
@@ -39,3 +39,35 @@ Keyset\keys(dict['x' => 1]); // keyset['x']
 
 Vec\map(keyset[1, 2], $x ==> $x + 1); // vec[2, 3]
 ```
+
+## Arrays Cheat Sheet
+
+| Operation| `vec`    | `dict`   | `keyset` |
+|----------|----------|----------|----------|
+| Initialize empty                             | `$v = vec[];`                | `$d = dict[];`                 | `$k = keyset[];`               |
+| Literal                                      | `$v = vec[1, 2, 3];`         | `$d = dict['foo' => 1];`       | `$k = keyset['foo', 'bar'];`   |
+| From Another Container*                      | `$v = vec($container);`      | `$d = dict($keyed_container);` | `$k = keyset($container);`     |
+| Keys from Container*                         | `$v = Vec\keys($container);` | N/A                            | `$k = Keyset\keys($container);`|
+| Add Elements                                 | `$v[] = 4;`                  | `$d['baz'] = 2;`               | `$k[] = 'baz';`                |
+| Bulk Add Elements                            | `$v = Vec\concat($t1, $t2)`  | `$d = Dict\merge($kt1, $kt2)`  | `$k = Keyset\union($t1, $t2)`  |
+| Remove Elements                              | Remove-at-index is unsupported; `$first = PHP\array_shift($v)` and `$last = PHP\array_pop($v)` | `unset($d['baz']);`| `unset($k['baz']);`|
```

Comment:
`Vec\drop()` and `Vec\take()` for Vecs. `PHP\` only exists at Facebook and should be removed from here

Reply:
The calling convention of `PHP\array_shift` and `PHP\array_pop` is by inout. Would be nice to add the `inout` keyword before the argument name. Also, unless you need the TAny permissive typing of the hhi of these functions, the hsl offers `HH\Lib\C\pop_back(x)` and `HH\Lib\C\pop_front(x)`. Let's encourage typesafe apis.

Reply:
if I understand right, looks like `shift` and `pop` should be removed in favor of `pop_back` and `pop_front`; (ignore whitespace; fixing that) like this?
![image](https://user-images.githubusercontent.com/5179225/127912594-c4087242-d598-4eda-a000-fadc4c5ca724.png)

Reply:
Yes, this is correct use of pop back and pop front.